esp/templates/program/modules/resourcemodule/classrooms.html

```diff
@@ -74,3 +97,71 @@ <h2>Step 3: Classrooms</h2>
 {% endifequal %}
 
 <br />
+
+<!--This branch has a fix for when the form is originally hidden-->
+<script src="https://rawgit.com/elo80ka/django-dynamic-formset/f5f68c818953ddf43ffa7cb2e9aa391c3d4dfb47/src/jquery.formset.js"></script>
+<!--jQuery.initialize plugin is created to help maintain dynamically created elements on the page-->
+<script src="https://rawgit.com/pie6k/jquery.initialize/master/jquery.initialize.js"></script>
+<script>
+<!--Formset settings-->
+$j('.furnishing-formset').formset({
+    prefix: 'furnishings',
+    formTemplate: '#id_empty_form',
+    addText: 'add furnishing',
+    deleteText: 'remove',
+    addCssClass: 'furnishing-add',
+    deleteCssClass: 'furnishing-delete',
+    formCssClass: 'furnishing-dynamic-form',
+});
+
+<!--Get choices for a furnishing via POST-->
+function getChoices(furnishing, callback){
+    var data = {furnishing: furnishing, csrfmiddlewaretoken: csrf_token()};
+    $j.post('/manage/{{ prog.url }}/ajaxfurnishingchoices', data, "json").success(callback);
+}
+
+<!--Update the option field to reflect whether there are specific choices (dropdown) or not (open response). If a value is supplied, set that value as selected afterwards.-->
+function update_choices(obj, value) {
+    var furnishing = $j(obj).val()
+    getChoices(furnishing, function(response) {
+        var choices = response.choices;
+        var num = $j(obj).attr('id').split("-")[1];
+        $j('#id_furnishings-' + num + '-choice').remove();
+        if (choices.length > 1) {
+            $j(obj).after('<select style="margin-left: 4px" id="id_furnishings-' + num + '-choice" name="furnishings-' + num + '-choice" type="text"></select>');
+            $j('#id_furnishings-' + num + '-choice').append('<option>(option)</option>');
+            for (i in choices) {
+                $j('#id_furnishings-' + num + '-choice').append('<option value=' + choices[i] + '>' + choices[i] + '</option>');
```

Comment:
Doing the string interpolation like this is a little questionable security-wise -- maybe it's not a problem since only admins can edit these, but better would be to either escape it properly, or use jquery's methods to populate it (e.g. I think you can do something like `...append($('<option>').value(choices[i]).text(choices[i]))`, although I don't recall the syntax offhand).  (You could also ideally do that for `num`, although it may be easier to just ensure it's an integer instead.)

Reply:
Good point, it's also extremely easy to mess up because you need to keep track of the different quote types!